examples/example_weather_worker.py

```diff
@@ -0,0 +1,234 @@
+)
+logger = logging.getLogger(__name__)
+
+def get_weather(city: str) -> Dict[str, Any]:
+    """Get weather information for a given city.
+    
+    This function makes an API call to fetch current weather data and provides
+    clothing recommendations based on the temperature and conditions.
+    
+    Args:
+        city (str): Name of the city to get weather for
+        
+    Returns:
+        Dict[str, Any]: Weather information including:
+            - status: API call status ('success' or 'error')
+            - message: Human-readable weather description
+            - data: Detailed weather information including:
+                - city: City name
+                - temperature: Temperature in Celsius
+                - condition: Weather condition (e.g., 'Sunny', 'Cloudy')
+                - humidity: Humidity percentage
+                - clothing: Recommended clothing based on conditions
+            
+    Raises:
+        requests.RequestException: If the API call fails
+        
+    Example:
+        >>> result = get_weather("New York")
+        >>> print(result["data"]["clothing"])
+        'Light jacket'
+    """
+    try:
+        # Make API call (using mock API for example)
+        response = requests.get('https://dylanburkey.com/assets/weather.json')
+        response.raise_for_status()
+        
+        # Process weather data (mock data for example)
+        weather_data = {
+            'New York': {'temp': '15°C', 'condition': 'Sunny', 'humidity': '60%'},
+            'Miami': {'temp': '28°C', 'condition': 'Sunny', 'humidity': '70%'},
+            'Boston': {'temp': '10°C', 'condition': 'Cloudy', 'humidity': '65%'}
+        }.get(city, {'temp': '20°C', 'condition': 'Clear', 'humidity': '50%'})
+        
+        # Get temperature as number for clothing recommendation
+        temp = int(weather_data['temp'].rstrip('°C'))
+        
+        # Determine appropriate clothing
+        if temp > 25:
+            clothing = 'Shorts and t-shirt'
+        elif temp > 15:
+            clothing = 'Light jacket'
+        else:
+            clothing = 'Sweater'
+            
+        return {
+            'status': 'success',
+            'message': f"Current weather in {city}: {weather_data['condition']}, "
+                      f"{weather_data['temp']}°F, {weather_data['humidity']} humidity",
+            'data': {
+                'city': city,
+                'temperature': weather_data['temp'],
+                'condition': weather_data['condition'],
+                'humidity': weather_data['humidity'],
+                'clothing': clothing
+            }
+        }
+        
+    except requests.RequestException as e:
+        logger.error(f"Failed to get weather: {e}")
+        return {
+            'status': 'error',
+            'message': f"Failed to get weather for {city}",
+            'data': None
+        }
+
+def get_worker_state(function_result: Optional[Dict], current_state: Optional[Dict]) -> Dict[str, Any]:
```

Comment:
I wonder if this should be named `get_and_update_worker_state()` to reflect that it doesn't just get the state, it also updates it.